### PR TITLE
Use virtual table to improve render efficiency

### PR DESCRIPTION
**Pull request by cw75**
## Describe your changes and why you are making these changes
This PR replaces our `DataTable` component with the React virtualized table that only renders the rows visible on the browser, which significantly improves the rendering performance. See the demo below:

https://www.loom.com/share/f0eab6f015994c36be39cc13e074fd3f

@agiron123 we may need to add the `--force` flag for the UI build test to pass, due to the peer dependency issue.
## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: scripts/install_local.py
Comment (vsreekanti) [inline scripts/install_local.py:121]: Hm, is this something we want to do every time? That means that running `install_local` is going to be a pretty expensive operation.

File: scripts/install_local.py
Comment (vsreekanti) [inline scripts/install_local.py:124]: I'm also not sure if this script will work if we're not actually building the `common` lib.

File: src/ui/common/src/components/tables/virtualizedTable.tsx
Comment (vsreekanti) [inline src/ui/common/src/components/tables/virtualizedTable.tsx:14]: Are these not exposed as constants from `react-virtualized`? I know MUI does this. It'd make it more maintainable for the future (i.e., if these class names change).

File: src/ui/common/src/components/tables/virtualizedTable.tsx
Comment (vsreekanti) [inline src/ui/common/src/components/tables/virtualizedTable.tsx:5]: Nit: We've been doing `import React from "react";` by default. Slightly cleaner to read.

File: src/ui/common/src/components/tables/virtualizedTable.tsx
Comment (vsreekanti) [inline src/ui/common/src/components/tables/virtualizedTable.tsx:28]: Should we at least make these class names `const`s for the time being as well?

File: src/ui/common/src/components/tables/virtualizedTable.tsx
Comment (vsreekanti) [inline src/ui/common/src/components/tables/virtualizedTable.tsx:99]: Is there a reason we're doing a class component here as opposed to a functional component? Also, what is a `PureComponent`? @agiron123, I know you'd like us to use FCs everywhere, so feel free to chime in.

Comment (vsreekanti) [inline src/ui/common/src/components/tables/virtualizedTable.tsx:99]: Style nit: The fact that the table is virtualized seems like an implementation detail, so I think we should still call this a `DataTable`. 

Also, I think the general style we've been trying to follow is that both the class names and file names should be PascalCased. So here, I think we should call both `DataTable`.

File: src/ui/common/src/components/integrations/integrationObjectPreview.tsx
Comment (vsreekanti) [inline src/ui/common/src/components/integrations/integrationObjectPreview.tsx:27]: Style nit: These blocks could be spaced a little better.

File: src/ui/common/src/components/integrations/integrationObjectPreview.tsx
Comment (vsreekanti) [inline src/ui/common/src/components/integrations/integrationObjectPreview.tsx:30]: Do we actually expect that refreshing the page will help anything here? If not, we shouldn't say that. 🙂

File: src/ui/common/src/components/integrations/integrationObjectPreview.tsx
Comment (vsreekanti) [inline src/ui/common/src/components/integrations/integrationObjectPreview.tsx:32]: Let's get in the habit of putting our error messages in `<pre>` blocks so that they format in a readable fashion.

File: src/ui/common/src/components/integrations/integrationObjectPreview.tsx
Comment (vsreekanti) [inline src/ui/common/src/components/integrations/integrationObjectPreview.tsx:31]: Rather than adding a `<br />` here, lets instead put the above title in an `<AlertTitle>` component and then have the error message be the body of the error.

**Review by vsreekanti — COMMENTED** [review]
This is *awesome*, @cw75! The demo looks really good. I know this wasn't easy, so thanks for soldiering through the vagaries of third-party React libraries. I've left a few style nits in line below. 

The only thing I'm not sure about is whether we want to (or even can?) make the new table component a functional component. @agiron123 might know better than me here, so I'll let him comment before we move forward. Other than that, I think this is 💯.

File: scripts/install_local.py
Comment (cw75) [reply scripts/install_local.py:121]: @agiron123

File: scripts/install_local.py
Comment (cw75) [reply scripts/install_local.py:124]: @agiron123

File: src/ui/common/src/components/integrations/integrationObjectPreview.tsx
Comment (cw75) [reply src/ui/common/src/components/integrations/integrationObjectPreview.tsx:30]: @likawind

File: src/ui/common/src/components/tables/virtualizedTable.tsx
Comment (cw75) [reply src/ui/common/src/components/tables/virtualizedTable.tsx:14]: I don't think so after poking around a bit. The example code on the web also have these class name strings hard-coded..

File: src/ui/common/src/components/tables/virtualizedTable.tsx
Comment (cw75) [reply src/ui/common/src/components/tables/virtualizedTable.tsx]: At the end of this file we have: `const DataTable = styled(<ClassName>)(styles);`. To prevent name collision I'm naming the class `PreStyledDataTable`. Let me know if you have better ideas. @agiron123 @vsreekanti

Comment (cw75) [reply src/ui/common/src/components/tables/virtualizedTable.tsx]: I believe this is from a code template we copied over. @agiron123 I'll let you figure out if we should do sth else here. :)

**Comment by cw75** [comment]
Addressed comments. @likawind I think some of @vsreekanti 's comments are on your changes, so would be nice if you can take a quick look as well.

File: src/ui/common/src/components/tables/virtualizedTable.tsx
Comment (vsreekanti) [reply src/ui/common/src/components/tables/virtualizedTable.tsx]: That's fine with me!

**Comment by vsreekanti** [comment]
Awesome! I'll wait for @agiron123 to weigh in on the functional component thing, and we should be good to go.

File: scripts/install_local.py
Comment (agiron123) [reply scripts/install_local.py:124]: @vsreekanti @cw75 ```npm install``` when run inside common actually builds common at the end. This is a redundant command that will save us a couple minutes on build time.

File: scripts/install_local.py
Comment (agiron123) [reply scripts/install_local.py:121]: Technically correct, it will take longer. This ensures that people have a clean build and that things should work properly.

We can also add a flag to allow people to choose whether or not they want to do a clean build. Personally I'm of the camp of doing clean builds every time as there's less moving parts and things to worry about.

In the future I'd like to look into some other build tools to use. I'm pretty sure we can get much faster build times by adopting a different build tool for common.

File: src/ui/common/src/components/tables/virtualizedTable.tsx
Comment (agiron123) [reply src/ui/common/src/components/tables/virtualizedTable.tsx]: styles is a const, so I think it's fine to leave them as is.

My suggestion here would be to move the styles into a file of their own.
so we'd have virtualizedTable.tsx and virtualizedTable.styles.tsx.

Then the styles can be exported and shared amongst other components. I'd also be sure to export the styles from common (src/index.ts) as well. If we haven't done so already, the VirtualTable component should be in there as well.

File: src/ui/common/src/components/tables/virtualizedTable.tsx
Comment (agiron123) [reply src/ui/common/src/components/tables/virtualizedTable.tsx]: @vsreekanti as @cw75 mentioned, this was part of the example code given by MUI. All of the other examples in MUI use functional components. Pure components allow us to have more control over when to re-render a particular component when it's parents update.

See this blog post here for a quick rundown: 
https://dev.to/ashwin_mothilal/difference-between-pure-component-and-functional-component-51mi

If we wish to use a functional component, we can probably do so using the useMemo() hook. I don't really have an opinion either way, we can just leave as is now since things are working quite well. I'd rather use what the example did on MUI, since they probably know more about how MUI and react-virtualized work together than we do :).

Comment (agiron123) [reply src/ui/common/src/components/tables/virtualizedTable.tsx]: @vsreekanti @cw75 I'd rather leave this one named VirtualizedTable and have DataTable be the parent class that all other tables inherit from. (more of an OOP analogy here, but you get the point)

WIth that being said, I have a table view that I am working on right now that will have pagination. 

After that goes in, we can combine all three into one table component and control which table component is used based on the props that we pass in.

let's leave as is for now and then we can revisit and build out a nicer DataTable implementation afterwards.

File: src/ui/app/package.json
Comment (agiron123) [inline src/ui/app/package.json:13]: please npm publish after this PR gets merged in.

File: src/ui/common/src/components/integrations/integrationObjectPreview.tsx
Comment (agiron123) [reply src/ui/common/src/components/integrations/integrationObjectPreview.tsx]: As a general rule, we really shouldn't be using <br /> and should be using css margins/paddings instead.

File: src/ui/common/src/components/tables/virtualizedTable.tsx
Comment (agiron123) [reply src/ui/common/src/components/tables/virtualizedTable.tsx]: I think this just came from the example code that we brought over, but should be an easy fix to stay consistent.

File: src/ui/common/src/components/tables/virtualizedTable.tsx
Comment (agiron123) [inline src/ui/common/src/components/tables/virtualizedTable.tsx]: ===

Comment (agiron123) [reply src/ui/common/src/components/tables/virtualizedTable.tsx]: However, JS is truthy, so we can do this instead:
```
if (!column.columnWidth)
```

**Review by agiron123 — APPROVED** [review]
Good to merge once comments are resolved.

File: src/ui/common/src/components/integrations/integrationObjectPreview.tsx
Comment (likawind) [inline src/ui/common/src/components/integrations/integrationObjectPreview.tsx:31]: Yeah we should remove the page refreshing thing

**Pull request merged**
